# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build util-macros-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure (ac-2.63; am-1.11; lt-2.2.6)
libtoolize -c -f; aclocal -I m4; autoheader; autoconf; automake -a -c
./configure --disable-static --prefix=

# build
make

# install
install -d $DESTDIR/include/X11/fonts
install -m 644 include/X11/fonts/fontenc.h $DESTDIR/include/X11/fonts
install -d $DESTDIR/lib
install src/.libs/libfontenc.so.1.* $DESTDIR/lib/libfontenc.so.1
ln -sf libfontenc.so.1 $DESTDIR/lib/libfontenc.so
install -d $DESTDIR/lib/pkgconfig
install -m 644 fontenc.pc $DESTDIR/lib/pkgconfig

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/libfontenc-build.pc << EOF
Name: libfontenc
Version: 1.0.5
Description: font encoding library
Requires: $REQUIRES

devel=\\
/include/X11/fonts/fontenc.h \\
/lib/libfontenc.so \\
/lib/pkgconfig/fontenc.pc

exec=\\
/lib/libfontenc.so.1
EOF

exit

# patch
--- libfontenc-1.0.5.orig/configure.ac
+++ libfontenc-1.0.5/configure.ac
@@ -24,6 +24,7 @@
 AC_PREREQ([2.57])
 
 AC_INIT(libfontenc, 1.0.5, [https://bugs.freedesktop.org/enter_bug.cgi?product=xorg], libfontenc)
+AC_CONFIG_MACRO_DIR([m4])
 AM_INIT_AUTOMAKE([dist-bzip2])
 AM_MAINTAINER_MODE
 
@@ -47,12 +48,6 @@
 
 AC_CHECK_LIB(z, gzclose) 
 
-# Check for dependencies
-PKG_CHECK_MODULES(FONTENC, xproto)
-FONTENC_CFLAGS="$CWARNFLAGS $FONTENC_CFLAGS"
-AC_SUBST(FONTENC_CFLAGS)
-AC_SUBST(FONTENC_LIBS)
-
 dnl Allow checking code with lint, sparse, etc.
 XORG_WITH_LINT
 LINT_FLAGS="${LINT_FLAGS} ${FONTENC_CFLAGS}"
--- libfontenc-1.0.5.orig/src/Makefile.am
+++ libfontenc-1.0.5/src/Makefile.am
@@ -5,8 +5,6 @@
          fontenc.c \
 	 fontencI.h
 
-AM_CFLAGS = @FONTENC_CFLAGS@
-
 FONTENCDIR=@ENCODINGSDIR@
 FONTENCDEFS = -DFONT_ENCODINGS_DIRECTORY=\"$(FONTENCDIR)/encodings.dir\"
 
@@ -14,8 +12,6 @@
 	   -DFONTENC_NO_LIBFONT $(FONTENCDEFS)
 
 
-libfontenc_la_LIBADD = @FONTENC_LIBS@
-
 libfontenc_la_LDFLAGS = -version-number 1:0:0 -no-undefined
 
 libfontencincludedir = $(includedir)/X11/fonts

# source
https://x.org/releases/individual/lib/libfontenc-1.0.5.tar.gz
